packages/create-block/lib/templates.js

```diff
@@ -1,91 +1,89 @@
+/**
+ * External dependencies
+ */
+const glob = require( 'fast-glob' );
+const { readFile } = require( 'fs' ).promises;
+const { fromPairs } = require( 'lodash' );
+const { join } = require( 'path' );
+
 /**
  * Internal dependencies
  */
 const CLIError = require( './cli-error' );
 const prompts = require( './prompts' );
 
-const namespace = 'create-block';
-const dashicon = 'smiley';
-const category = 'widgets';
-const author = 'The WordPress Contributors';
-const license = 'GPL-2.0-or-later';
-const licenseURI = 'https://www.gnu.org/licenses/gpl-2.0.html';
-const version = '0.1.0';
-
-const templates = {
+const predefinedBlockTemplates = {
 	es5: {
 		defaultValues: {
-			namespace,
 			slug: 'es5-example',
 			title: 'ES5 Example',
 			description:
 				'Example block written with ES5 standard and no JSX – no build step required.',
-			dashicon,
-			category,
-			author,
-			license,
-			licenseURI,
-			version,
 		},
-		outputFiles: [
-			'.editorconfig',
-			'editor.css',
-			'index.js',
-			'$slug.php',
-			'style.css',
-			'readme.txt',
-		],
+		wpScripts: false,
 	},
 	esnext: {
 		defaultValues: {
-			namespace,
 			slug: 'esnext-example',
 			title: 'ESNext Example',
 			description:
 				'Example block written with ESNext standard and JSX support – build step required.',
-			dashicon,
-			category,
-			author,
-			license,
-			licenseURI,
-			version,
 		},
-		outputFiles: [
-			'.editorconfig',
-			'.gitignore',
-			'editor.css',
-			'src/edit.js',
-			'src/index.js',
-			'src/save.js',
-			'$slug.php',
-			'style.css',
-			'readme.txt',
-		],
-		wpScriptsEnabled: true,
 	},
 };
 
-const getTemplate = ( templateName ) => {
-	if ( ! templates[ templateName ] ) {
+const getOutputTemplates = async ( name ) => {
+	const outputTemplatesPath = join( __dirname, 'templates', name );
+	const outputTemplatesFiles = await glob( '**/*.mustache', {
+		cwd: outputTemplatesPath,
+		dot: true,
+	} );
+	return fromPairs(
+		await Promise.all(
+			outputTemplatesFiles.map( async ( outputTemplateFile ) => {
+				const outputFile = outputTemplateFile.replace(
+					'.mustache',
```

Comment:
If we're expecting it to be a `.mustache` file extension, we can give ourselves a bit more confidence by ensuring that the replacement only occurs at the end of the file name:

```suggestion
					/\.mustache$/,
```